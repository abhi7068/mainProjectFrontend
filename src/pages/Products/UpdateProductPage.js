import React, { Component } from 'react';
import UpdateProduct from '../../organisms/Products/UpdateProduct';
import PropTypes from 'prop-types';

export default class UpdateProductPage extends Component {
  render() {
    const {
      match: {
        params: { id },
      },
    } = this.props;
    return (
      <div>
        <UpdateProduct id={id} />
      </div>
    );
  }
}
UpdateProductPage.propTypes = {
  match: PropTypes.object,
};
