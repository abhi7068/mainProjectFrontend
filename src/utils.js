import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
