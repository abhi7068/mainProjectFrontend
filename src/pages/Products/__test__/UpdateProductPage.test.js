import React from 'react';

import UpdateProductPage from '../UpdateProductPage';
import { render } from '../../../utils';

it('it renders', () => {
  render(<UpdateProductPage match={{ params: '' }} />);
});
it('should take a snapshot', () => {
  const { asFragment } = render(<UpdateProductPage match={{ params: '' }} />);
  expect(
    asFragment(render(<UpdateProductPage match={{ params: '' }} />))
  ).toMatchSnapshot();
});
