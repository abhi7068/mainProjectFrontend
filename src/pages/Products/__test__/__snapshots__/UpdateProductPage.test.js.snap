// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="MuiPaper-root MuiCard-root index-root-211 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
            >
              UPDATE PRODUCT
            </span>
          </div>
        </div>
        <form>
          <div>
            <div>
              <legend
                class="MuiFormLabel-root"
                data-testid="label"
              >
                Product Name
              </legend>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="width: 225px;"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-basic"
                  name="product_name"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-326 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-327"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <br />
            <div>
              <legend
                class="MuiFormLabel-root"
                data-testid="label"
              >
                Description
              </legend>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="width: 225px;"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-basic"
                  name="description"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-326 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-327"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <br />
            <div>
              <legend
                class="MuiFormLabel-root"
                data-testid="label"
              >
                Price
              </legend>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="width: 225px;"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-basic"
                  name="price"
                  required=""
                  type="Number"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-326 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-327"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <br />
            <div>
              <legend
                class="MuiFormLabel-root"
                data-testid="label"
              >
                Discount
              </legend>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="width: 225px;"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-basic"
                  max="100"
                  min="1"
                  name="discount"
                  type="Number"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-326 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-327"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root index-formControl-214"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
              id="demo-simple-select-label"
            >
              Category
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby=" mui-component-select-category"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="mui-component-select-category"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                name="category"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <br />
          <br />
          <div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="button"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
